core/schema/sdk.go

```diff
@@ -932,3 +1097,132 @@ func (sdk *goSDK) base(ctx context.Context) (dagql.Instance[*core.Container], er
 	}
 	return ctr, nil
 }
+
+func gitConfigSelectors(ctx context.Context, bk *buildkit.Client) ([]dagql.Selector, error) {
+	// codegen runs `go mod tidy` and for private deps
+	// we allow users to configure GOPRIVATE env variable.
+	// But for it to work, we need to ensure we don't run into
+	// host checking prompt. So customizing GIT_SSH_COMMAND to
+	// allow skipping the prompt.
+	selectors := []dagql.Selector{
+		{
+			Field: "withEnvVariable",
+			Args: []dagql.NamedInput{
+				{
+					Name:  "name",
+					Value: dagql.NewString("GIT_SSH_COMMAND"),
+				},
+				{
+					Name:  "value",
+					Value: dagql.NewString("ssh -o StrictHostKeyChecking=no "),
+				},
+			},
+		},
+	}
+
+	gitconfig, err := bk.GetGitConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range gitconfig {
```

Comment:
Is there a way to avoid this being in lots of little `WithExec`s? I think I mentioned this before in https://github.com/dagger/dagger/pull/9323#discussion_r1950646188.

Reply:
I did tried it. It seems like not... (unless we take complete git config as is).  once we fetch config using `git config -l`, we cannot translate it back to a working gitconfig file unless we call `git config` on those retrieved values again.

Reply:
:thinking: is it not just as simple as appending to the gitconfig file manually? Or is that just too fundamentally complex?

The other option would be to combine all of these by passing them to a `sh` and joining them with `&&`.

Reply:
Potentially https://github.com/go-git/go-git/blob/main/plumbing/format/config/common.go might also be useful here?

Reply:
yep, this seems useful to me.

Reply:
i tried this one. it seems it does not support merging git configs (e.g. system level vs user level vs repo level).

although when running `git config -l` through session attachable, it does not consider repo level config either.

another thing to notice is that we have a safe list of git config keys allowed right now, which is just one, so having withExec is not going to be a problem unless we allow a large number of keys (or remove that check) from here.

Reply:
Ah, yes, I'd forgotten we only allow the url transformation.